system/Helpers/array_helper.php

```diff
@@ -218,3 +218,49 @@ function array_flatten_with_dots(iterable $array, string $id = ''): array
         return $flattened;
     }
 }
+
+if (! function_exists('array_group_by')) {
+    /**
+     * Groups all rows by their index values. Result's depth equals number of indexes
+     *
+     * @param array $array        Data array (i.e. from query result)
+     * @param array $indexes      Indexes to group by. Dot syntax used. Returns $array if empty
+     * @param bool  $includeEmpty If true, null and '' are also added as valid keys to group
+     *
+     * @return array Result array where rows are grouped together by indexes values.
+     */
+    function array_group_by(array $array, array $indexes, bool $includeEmpty = false): array
+    {
+        if (empty($indexes)) {
```

Comment:
Use strict comparison:
```suggestion
        if ($indexes === []) {
```

Reply:
Makes sense.